File: kvm/ui.py
from enum import Enum

import shared
from shared import get_image_path

IMAGES_DIR = shared.get_images_dir(__file__)


class ImagesMapping(str, Enum):
    desktop = get_image_path(IMAGES_DIR, "KVM_desktop.bmp")
    mac_mini = get_image_path(IMAGES_DIR, "KVM_mac_mini.bmp")
    work_mac = get_image_path(IMAGES_DIR, "KVM_work_mac.bmp")
    error = get_image_path(IMAGES_DIR, "Error_kvm.bmp")

File: v2/kvm/ui.go
package kvm

import (
	"embed"
	"github.com/nathan-coates/desk_controller/v2/shared"
	"log"
)

//go:embed *.bmp
var images embed.FS

var (
	KvmDesktop string
	KvmMacMini string
	KvmWorkMac string
	KvmError   string
)

func init() {
	mapping, err := shared.WriteImagesToFile(images)
	if err != nil {
		log.Fatal(err)
	}

	KvmDesktop = mapping["KVM_desktop.bmp"]
	KvmMacMini = mapping["KVM_mac_mini.bmp"]
	KvmWorkMac = mapping["KVM_work_mac.bmp"]
	KvmError = mapping["Error_kvm.bmp"]
}
